functional/FunctionalTest.java

```diff
@@ -327,6 +390,152 @@ public static void getObject_test5() throws Exception {
   }
 
   /**
+   * Test: getObject(String bucketName, String objectName, EncryptionMaterials encryptionMaterials).
+   */
+  public static void getObject_test6() throws Exception {
+    System.out.println(
+        "Test: client side encryption (AES 256 bit): getObject(String bucketName, String objectName, "
+            + "EncryptionMaterials encryptionMaterials).");
+
+    String fileName = createFile(13 * MB);
+    InputStream is = Files.newInputStream(Paths.get(fileName));
+
+    // Generate symmetric 128 bit AES key.
+    KeyGenerator symKeyGenerator = KeyGenerator.getInstance("AES");
+    symKeyGenerator.init(128);
+    SecretKey symKey = symKeyGenerator.generateKey();
+
+    try {
+      // Put an encrypted object
+      client.putObject(bucketName, fileName, is, 20 * 1024 * 1024, null, symKey);
+    } catch (InsufficientDataException e) {
+      ignore();
+    }
+    is.close();
+
+    // Get the object without decryption
+    String plainFileName = createFile(0);
+    InputStream ois = client.getObject(bucketName, fileName);
+    OutputStream os = Files.newOutputStream(Paths.get(plainFileName));
+
+    // Read in the decrypted bytes and write to fileNameOut
+    int numRead = 0;
+    byte[] buf = new byte[8192];
+    while ((numRead = ois.read(buf)) >= 0) {
+      os.write(buf, 0, numRead);
+    }
+    os.close();
+    ois.close();
+
+    // Check if two files are not equal
+    File file1 = new File(fileName);
+    File file2 = new File(plainFileName);
+
+    if (FileUtils.contentEquals(file1, file2)) {
+      throw new MinioException("Files should not be equal");
+    }
```

Comment:
You can use like below.  Test is not that critical to be optimized.

```java
    if (!Arrays.equals(Files.readAllBytes(Paths.get(filename)),
                       Files.readAllBytes(Paths.get(filename + ".downloaded")))) {
      throw new Exception("[FAILED] Test: presignedGetObject(String bucketName, String objectName)"
                          + ", Error: <Content differs>");
    }
```

You can refer `presignedGetObject_test1()`, how things are done simpler.